metecho/api/serializers.py

```diff
@@ -317,8 +354,10 @@ def validate_github_users(self, github_users):
 class TaskSerializer(serializers.ModelSerializer):
     id = serializers.CharField(read_only=True)
     description_rendered = MarkdownField(source="description", read_only=True)
-    epic = serializers.PrimaryKeyRelatedField(
-        queryset=Epic.objects.all(), pk_field=serializers.CharField()
+    epic = NestedPrimaryKeyRelatedField(
+        queryset=Epic.objects.all(),
+        pk_field=serializers.CharField(),
+        serializer=EpicMinimalSerializer,
```

Comment:
@oluoluoxenfree With this change, `task.epic` is now an object with nested data instead of a string:

```
epic: {
  id: string;
  name: string;
  slug: string;
  github_users: string[];
}
```

This will require updating the type definition in `tasks/reducer.ts` (https://github.com/oddbird/Metecho/blob/view-all-tasks/src/js/store/tasks/reducer.ts#L29), and then making corresponding changes anywhere `task.epic` is being accessed in the JS.

But it should provide the info you need for the table (`name` to display in the new "Epic" column, and `slug` to construct the task URL), as well as the "Epic Collaborators" list in the `assignTaskRole` modal. For the latter, see the logic in `useFetchEpicIfMissing.tsx` (https://github.com/oddbird/Metecho/blob/view-all-tasks/src/js/components/utils/useFetchEpicIfMissing.tsx#L24-L26) for an example of how to convert `task.epic.github_users` into full user objects (instead of `id` strings).